library/network_connections.py

```diff
@@ -359,16 +360,37 @@ def ifcfg_create(
         if connection["mtu"]:
             ifcfg["MTU"] = str(connection["mtu"])
 
+        ethtool_options = ""
         if "ethernet" in connection:
             if connection["ethernet"]["autoneg"] is not None:
                 if connection["ethernet"]["autoneg"]:
-                    s = "autoneg on"
+                    ethtool_options = "autoneg on"
                 else:
-                    s = "autoneg off speed %s duplex %s" % (
+                    ethtool_options = "autoneg off speed %s duplex %s" % (
                         connection["ethernet"]["speed"],
                         connection["ethernet"]["duplex"],
                     )
-                ifcfg["ETHTOOL_OPTS"] = s
+
+        ethtool_features = connection["ethtool"]["features"]
+        configured_features = []
+        for feature, setting in ethtool_features.items():
+            value = ""
+            if setting:
+                value = "on"
+            elif setting is not None:
+                value = "off"
+
+            if value:
+                configured_features.append("%s %s" % (feature, value))
+
+        if configured_features:
+            ethtool_options += "-K %s %s" % (
```

Comment:
I think this misses a space when you combine it with duplex/speed options above.

Reply:
will this ETHTOOL_OPTS line actually work when passing to ethtool?


See `/etc/sysconfig/network-scripts/network-functions`
```
    IFS=';';
    if [ -n "${ETHTOOL_DELAY}" ]; then
      # Convert microseconds to seconds:
      local ETHTOOL_DELAY_SEC=$(convert2sec ${ETHTOOL_DELAY} micro)
      sleep ${ETHTOOL_DELAY_SEC}
    fi
    for opts in $ETHTOOL_OPTS ; do
        IFS=$oldifs;
        if [[ "${opts}" =~ [[:space:]]*- ]]; then
            /sbin/ethtool $opts
        else
            /sbin/ethtool -s ${REALDEVICE} $opts
        fi
        IFS=';';
    done
```


If you combine dulex/speed options, you need a ';'

Reply:
yes, thank you, fixed it.